postgres/src/test/java/org/jdbi/v3/postgres/TestInetAddressPg.java

```diff
@@ -13,19 +13,79 @@
  */
 package org.jdbi.v3.postgres;
 
+import java.net.Inet4Address;
+import java.net.InetAddress;
+
 import de.softwareforge.testing.postgres.junit5.EmbeddedPgExtension;
 import de.softwareforge.testing.postgres.junit5.MultiDatabaseBuilder;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.Sql;
 import org.jdbi.v3.core.argument.TestInetAddressH2;
 import org.jdbi.v3.core.junit5.PgDatabaseExtension;
+import org.jdbi.v3.testing.junit5.JdbiExtension;
+import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
-public class TestInetAddressPg extends TestInetAddressH2 { { dbExtension = PgDatabaseExtension.instance(pg).withPlugin(new PostgresPlugin()); }
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestInetAddressPg extends TestInetAddressH2 {
+
+    {
+        dbExtension = PgDatabaseExtension.instance(pg).withPlugin(new PostgresPlugin());
+    }
 
     @RegisterExtension
     public static EmbeddedPgExtension pg = MultiDatabaseBuilder.instanceWithDefaults().build();
 
+    @RegisterExtension
+    public JdbiExtension pgExtension = JdbiExtension.postgres(pg).withPlugin(new PostgresPlugin());
+
     @Override
     protected String getInetType() {
         return "INET";
     }
+
+    @Test
+    public void testPgInetAddress() throws Exception {
+        String tblName = "inet_test";
+        String colName = "ipaddress";
+        Sql createTableSql = Sql.of(
+            "CREATE TABLE", tblName, "(",
+            colName, getInetType(),
+            ")");
+
+        try (Handle h = dbExtension.getJdbi().open()) {
+            h.execute(createTableSql);
+
+            InetAddress inetAddrIn = InetAddress.getByName("8.8.8.8");
+            assertThat(inetAddrIn).isInstanceOf(Inet4Address.class);
+
+            // insert IP address into column of type INET
+            int update1Count = h.createUpdate(Sql.of("INSERT INTO", tblName, "(", colName, ") VALUES (?)"))
+                .bind(0, inetAddrIn)
+                .execute();
+            assertThat(update1Count).isOne();
+
+            // read back the inserted IP address
+            InetAddress inetAddr1Out = h.createQuery(Sql.of("SELECT", colName, "FROM", tblName))
+                .mapTo(InetAddress.class)
+                .one();
+
+            // test for equality
+            assertThat(inetAddrIn).isEqualTo(inetAddr1Out);
+
+            // update the record to a null IP address
+            int update2Count = h.createUpdate(Sql.of("UPDATE", tblName, "SET", colName, "= ?"))
```

Comment:
I think I never used positional bindings, so I guess that works. It is somewhat brittle, the main selling point of Jdbi is using named bindings. :-)

Reply:
That slipped in from the original bug report. Will update to named bindings :)